### ISSUE TITLE
[bug] Server responds with a status of 404 if the URL path contains a dot

### ISSUE THREAD

**Issue by rmburg** [tauri-apps/tauri#8205]
### Describe the bug

When navigating to a URL containing a dot (.) character, the server responds with a status of 404 - not found.

### Reproduction

- Create new empty tauri app (tested with `npm create tauri-app` with React+npm+TS)
- `npm tauri dev`, open devtools (Ctrl+Shift+I)
- run `location.pathname = "/some_path"` -> works, returns index.html
- run `location.pathname = "/some.path"` -> 404
Example output:
![image](https://github.com/tauri-apps/tauri/assets/42248344/d0343537-ed53-4443-8cc6-43a884c84dc6)

### Expected behavior

URLs should not break if they contain a dot.

### Platform and versions

```text
[✔] Environment
    - OS: Arch Linux Rolling Release X64
    ✔ webkit2gtk-4.0: 2.42.1
    ✔ rsvg2: 2.57.0
    ✔ rustc: 1.73.0 (cc66ad468 2023-10-03)
    ✔ Cargo: 1.73.0 (9c4383fb5 2023-08-26)
    ✔ rustup: 1.26.0 (2023-05-04)
    ✔ Rust toolchain: stable-x86_64-unknown-linux-gnu (environment override by RUSTUP_TOOLCHAIN)
    - node: 21.1.0
    - yarn: 1.22.19
    - npm: 10.2.3

[-] Packages
    - tauri [RUST]: 1.5.2
    - tauri-build [RUST]: 1.5.0
    - wry [RUST]: 0.24.4
    - tao [RUST]: 0.16.5
    - @tauri-apps/api [NPM]: 1.5.1
    - @tauri-apps/cli [NPM]: 1.4.0 (outdated, latest: 1.5.6)

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

**Comment by rmburg** [comment]
I found out that if the dot character makes sense in the sense of path resolution, it does not cause a 404 response.
Examples:
- `/abc/../def` -> works, redirects to `/def`
- `/abc/.` -> works, redirects to `/abc`

**Comment by olivierlemasle** [comment]
This is not a Tauri bug: it seems to be caused by Vite.

In your example of the app created with `npm create tauri-app` with React+npm+TS, you can reproduce the issue **without Tauri**, by running `npm run dev` (dev mode) or `npm run build && npm run preview` (prod).

It seems to be https://github.com/vitejs/vite/issues/2415, which is fixed in Vite v5.0.0-beta but is not fixed in v4.x

**Comment by rmburg** [comment]
Thanks a lot, I didn't think this could be caused by vite.
